uaclient/api/u/pro/attach/magic/wait/v1.py

```diff
@@ -0,0 +1,85 @@
+import time
+
+from uaclient import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UAConfig
+from uaclient.contract import UAContractClient
+from uaclient.data_types import DataObject, Field, StringDataValue
+
+
+class MagicAttachWaitOptions(DataObject):
+    fields = [
+        Field("magic_token", StringDataValue),
+    ]
+
+    def __init__(self, magic_token: str):
+        self.magic_token = magic_token
+
+
+class MagicAttachWaitResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("_schema", StringDataValue),
+        Field("confirmation_code", StringDataValue),
+        Field("token", StringDataValue),
+        Field("expires", StringDataValue),
+        Field("user_email", StringDataValue),
+        Field("contract_id", StringDataValue),
+        Field("contract_token", StringDataValue),
+    ]
+
+    def __init__(
+        self,
+        _schema: str,
+        confirmation_code: str,
+        token: str,
+        expires: str,
+        user_email: str,
+        contract_id: str,
+        contract_token: str,
+    ):
+        self._schema = _schema
+        self.confirmation_code = confirmation_code
+        self.token = token
+        self.expires = expires
+        self.user_email = user_email
+        self.contract_id = contract_id
+        self.contract_token = contract_token
+
+
+def wait(
+    options: MagicAttachWaitOptions,
+    cfg: UAConfig,
+) -> MagicAttachWaitResult:
+    contract = UAContractClient(cfg)
+
+    while True:
```

Comment:
Should we have a `maximum_attempts` or similar? I'm afraid of relying on any external source - no matter how reliable, to break an infinite loop.

Reply:
That makes sense, we should define the default value for this check. It seems that the token has a lifespan of 10 minutes. With a delay of 10s, we could check `maximum_attempts < 70` just to get an extra minute there. What do you think ?